test/component_test.rb

```diff
@@ -114,6 +116,30 @@ class ComponentTest < ActiveSupport::TestCase
     assert_equal [], component.foos
   end
 
+  class TestValidationComponent < Components::Component
+    attribute :foo
+    validates :foo, presence: true
+  end
```

Comment:
I've been sticking the setup in each test case for now, just something I'm trying out :) Let's stick with that for now! So basically just move this into the test case, check the other test cases to see how.

Reply:
Primary I defined those classes separately to DRY up the code and dodge an active model validation issue with missing class names on anonymous classes. Since I managed to find a workaround for the missing names I used anonymous classes inside the tests.